luigi/contrib/spark.py

```diff
@@ -226,76 +215,160 @@ def track_progress(self, proc):
         return proc.returncode, final_state, app_id
 
 
-class Spark1xJob(luigi.Task):
-
-    num_executors = None
+class SparkSubmitTask(luigi.Task):
+    """
+    Template task for running a spark job
+
+    Strictly follows spark-submit usage::
+
+        Usage: spark-submit [options] <app jar | python file> [app options]
```

Comment:
It would be better to just say something like "See ``spark-submit -h`` for more information.". Or just add documentation link of the spark-submit script.